src/haversine.jl

```diff
@@ -33,11 +33,12 @@ function (dist::Haversine)(x, y)
     2 * dist.radius * asin( min(√a, one(a)) ) # take care of floating point errors
 end
 
-haversine(x, y, radius::Real = Float32(6371000)) = Haversine(radius)(x, y)
+haversine(x, y, radius=Float32(6371000)) = Haversine(radius)(x, y)
 
 @noinline haversine_error(dist) = throw(ArgumentError("expected both inputs to have length 2 in $dist distance"))
 
-
+_deg2rad(x::Real) = deg2rad(x)
+_deg2rad(x::Number) = x # targets unitful numbers
```

Comment:
Actually, couldn't Unitful define `deg2rad` when the unit is degree? Otherwise, what happens if somebody passes a radius e.g. in kg?

Reply:
I think I found a much better way to do it, thanks for persisting and pushing. ;-)

Reply:
Great! Maybe we should keep the `::Number` restriction?